api/envoy/api/v2/core/base.proto

```diff
@@ -113,6 +142,21 @@ message Node {
   // version of Envoy is being tested in a heterogeneous fleet. This will be set
   // by Envoy in management server RPCs.
   string build_version = 5;
+
+  // Free-form string that identifies the entity requesting config.
+  // E.g. "envoy" or "grpc"
+  string user_agent_name = 6;
+
+  // Free-form string that identifies the version of the entity requesting config.
+  // E.g. "1.12.2" or "abcd1234", or "SpecialEnvoyBuild"
+  string user_agent_version = 7;
```

Comment:
They are there for other xDS clients. E.g. there will be "grpc c++", etc.

Reply:
Is there any reason these need to be reported separately from `build_version`?  The gRPC implementations already report a user-agent-style string in `build_version` (which indicates both the fact that it's gRPC and the language and version), so the information should already be there.  But I'm concerned that pulling this out into a separate field might encourage management servers to react differently based on the type of the client, which is probably not a good idea.

Reply:
All the xDS clients are envoy though? Is there a scenario where there is some kind of xDS proxy? I don't really see why a management server would care what language the GRPC client is written in?

Reply:
We are in the process of adding xDS support to gRPC.  The idea is that a gRPC client will get its configuration from xDS so that the traffic does not need to go through Envoy.  So gRPC will be another xDS client.  My previous comments were in the context of that work.

Reply:
@markdroth we previously agreed in the Envoy stable API versioning doc that user agent made sense. The idea is that this is an escape hatch for things like patching over bugs etc. It's not intended to be the primary way a management server negotiates features.

Reply:
@jpeach there are other xDS clients even beyond Envoy and gRPC, take a look at https://github.com/citrix/citrix-istio-adaptor. Many organizations also have internal debug clients. With UDPA, we expect this to grow further.

Reply:
@htuch But doesn't the `build_version` string accomplish that?  Splitting this out into separate fields seems like it just provides duplicate information and makes it a little too easy for management server implementations to do something that should ideally not happen very often.

Reply:
I think we should deprecate `build_version`. The user agent and version as distinct fields was what was agreed to, providing more structure here makes sense. 

I'm not seeing that there is a material difference in the barrier to entry for depending on client implementation; I'm sure most folks know how to do `build_version.split(" ")` or the like.

Reply:
I think folks generally find it easier to match the entire content of a string than to go spelunking within it, especially if there's no consistent format for information inside of the string.  It's basically the same situation as browser user-agent strings; there's a lot of variation in what different clients specify there, which makes it at least slightly harder to do browser-specific hacks (although people do still do them).

In any case, I guess this doesn't make a huge difference, so I can live with this as-is.

Reply:
I've deprecated the `build_version` field in favor of the `user_agent_name` and `user_agent_version` fields.